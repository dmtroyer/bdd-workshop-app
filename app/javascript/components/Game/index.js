import React from 'react';
import WarGamesIntro from '../WarGamesIntro';

const Game = () => <WarGamesIntro />;

export default Game;
