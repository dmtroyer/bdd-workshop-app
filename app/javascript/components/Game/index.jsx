import React from 'react';
import { Redirect } from 'react-router-dom';
import { Spinner } from 'reactstrap';
import useProfile from '../../hooks/useProfile';
import MainContainer from '../MainContainer';
import GameProfile from '../Profile/GameProfile';

const Game = () => {
  const { profile, loading } = useProfile();

  if (loading) return <Spinner data-testid="game-spinner" />;
  if (profile === undefined) {
    return <Redirect to="/register" />;
  }
  return (
    <MainContainer dataTestId="game">
      <h1>coming soon</h1>
      <GameProfile profile={profile} />
    </MainContainer>
  );
};

export default Game;
