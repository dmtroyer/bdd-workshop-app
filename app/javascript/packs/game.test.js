import React from 'react';
import { configure, shallow } from 'enzyme';
import Game from './game';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Game', () => {
  it('renders a <div>', () => {
    const wrapper = shallow(<Game />);
    expect(wrapper.type()).toBe('div');
    expect(wrapper.prop('children')).toBe('16.7.0');
  });
});
