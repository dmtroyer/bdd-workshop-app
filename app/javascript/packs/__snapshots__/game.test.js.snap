// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders correctly 1`] = `
<div>
  <div
    data-test="message"
  >
    You are on React
  </div>
  <div>
    React Version: 
    <span
      data-test="react-version"
    >
      16.8.1
    </span>
  </div>
</div>
`;
